### PR TITLE
Implemented get existing Diagnostic Report

### PR DESCRIPTION
**Pull request by milankarunarathne**
It'll retrieve all required filed for Diagnostic Report.
@surangak @harsha89 plz, review the code!

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java [hunk 1]
```diff
@@ -27,20 +27,47 @@ public LaboratoryHandler() {
...
+
+		// @required: Get EncounterDateTime and set as `Issued` date
+		diagnosticReport.setIssued(new DateTimeDt(omrsiagnosticReport.getEncounterDatetime()));
+
+		// @required: Get Encounter Patient and set as `Subject`
+		org.openmrs.Patient omrsPatient = omrsiagnosticReport.getPatient();
+		diagnosticReport.getSubject().setResource(FHIRPatientUtil.generatePatient(omrsPatient));
+
 		// Get Encounter Provider and set as `Performer`
+		String encounterRoleUuid = "73bbb069-9781-4afc-a9d1-54b6b2270e03";
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java]: why it's hard coded?

Comment (milankarunarathne) [inline api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java]: It'll fix with [this ticket](https://issues.openmrs.org/browse/FM-160)

**Comment by milankarunarathne** [comment]
@harsha89 @surangak Can we merge this request, if there isn't any issue to be solve ?

File: omod/src/main/java/org/openmrs/module/fhir/providers/RestfulDiagnosticReportResourceProvider.java [hunk 2]
```diff
@@ -81,8 +81,7 @@ public MethodOutcome createFHIRDiagnosticReport(@ResourceParam DiagnosticReport
 	@Read()
 	public DiagnosticReport getResourceById(@IdParam IdDt theId) {
 		System.out.println("DiagnosticReport : getResourceByID");
-		DiagnosticReport result = null;
```
Comment (surangak) [inline omod/src/main/java/org/openmrs/module/fhir/providers/RestfulDiagnosticReportResourceProvider.java]: Sysouts ;)

File: api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java [hunk 3]
```diff
@@ -27,20 +27,47 @@ public LaboratoryHandler() {
...
 
-    public String getServiceCategory() {
-        return ServiceCategory;
-    }
+	public String getServiceCategory() {
+		return ServiceCategory;
+	}
+
+	@Override
+	public DiagnosticReport getFHIRDiagnosticReport(Encounter omrsiagnosticReport) {
```
Comment (surangak) [inline api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java]: I think this should receive a UUID, and not an encounter. Only the Handler itself knows that the UUID is for an encounter....

Comment (milankarunarathne) [inline api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java]: @surangak What if we have to search by `Name`, `Subject` etc ( [search parameters](http://hl7-fhir.github.io/diagnosticreport.html#4.21.6) ) ?
I think we also have to chagne method name into `getFHIRDiagnosticReportById(String id)`, is it ?

**Comment by surangak** [comment]
@milankarunarathne , I don't think we'll need name and subject, but for this, yeah, change to "ById" :)

**Pull request merged**